test/multiple-user-principal-types.test.js

```diff
@@ -373,19 +383,36 @@ describe('Multiple users with custom principalType', function() {
 
         return Album.create({name: 'album', userId: userFromOneModel.id})
           .then(function(album) {
-            return Role.isInRole(
-              Role.OWNER,
-              {
-                principalType: OneUser.modelName,
-                principalId: userFromOneModel.id,
-                model: Album,
-                id: album.id,
-              });
+            baseContext = {
+              principalId: userFromOneModel.id,
+              model: Album,
+              id: album.id,
+            };
+            var validContext = buildContextWith({principalType: OneUser.modelName});
+            var invalidContext = buildContextWith({principalType: AnotherUser.modelName});
+
+            return isInRole([
+              {role: Role.OWNER, context: validContext},
+              {role: Role.OWNER, context: invalidContext},
+            ]);
           })
-          .then(function(isInRole) {
-            expect(isInRole).to.be.true();
+          .then(function(data) {
+            expect(data).to.eql([true, false]);
           });
+
+        // helper
+        function buildContextWith(properties) {
+          return Object.assign(properties, baseContext);
+        }
       });
+
+      // helper
+      function isInRole(requests) {
+        var isInRoles =  requests.map(function(request) {
+          return Role.isInRole(request.role, request.context);
+        });
+        return Promise.all(isInRoles);
+      }
     });
 
```

Comment:
adjusted the test on built-in $owner role resolver to verify a user with similar id than an authorized user userOne but from another user model is denied access to album instance owned by userOne